### DISCUSSION TITLE
Switching to FXGL from own engine

### DISCUSSION THREAD

**Discussion by nicymike** [AlmasB/FXGL#1175] [Q&A] [answered]
Hi, I am Michael, the developer of Serflings which is basically a remake of the old game The Settlers 1 from BlueByte. I developed the game around my own engine with Java2D and Swing Canvas with custom rendering.

My plan is to switch to JavaFX and because I like the idea of FXGL, I decided to completely replace the game engine (challenges ahead :D ). I did some tests which raised some questions about how to do it with FXGL 11 (best practices).

1. When starting an empty game there is already stuff built in like mouse cursor, a menu with ESC and a blue image before the screen turns white. Do I have to disable all of this manually or is there a true plain mode available? What is the purpose of the blue image anyway?

2. My game supports multiple languages by using resource bundles. I already found a LocalizationService but what is the correct way to add the texts to fxgl?

3. I am currently saving the game settings (like window size or sound volume) to a json file. There is a SystemBundleService available but I did not find more information about it. How to save and load game settings within a file with fxgl or should I just continue to handle them myself?

I think this is it for now. Thank you for helping out. I think there will be more questions waiting if I start to dig deeper.

**Comment by AlmasB** [comment, marked as answer]
Hi Michael,

1. There are no different modes available for setting up the "aesthetics" of the application, though this sounds like a potential feature to be included. If you feel this has a general-purpose use case for it, opening an issue would help keep track of this. You can enable/disable/tweak most of these via the `GameSettings` object inside `initSettings()`.

2. Here's an [example](https://github.com/AlmasB/FXGL/blob/636eb63d0e40814d6823610655e26ff90d67f2a9/fxgl-samples/src/main/java/advanced/LocalizationSample.java) on how to do it from code. Though the API (from the same service) also allows you to load text from `.lang` files, which are essentially the same as a typical resource bundle.

3. If it already works for you, there is probably no strong reason to change it to FXGL's way of serializing data. Here's a [sample](https://github.com/AlmasB/FXGL/blob/636eb63d0e40814d6823610655e26ff90d67f2a9/fxgl-samples/src/main/java/advanced/SaveLoadSample.java) that shows how FXGL generally handles saving/loading.

Re: more questions, that's fine. The wiki and the `fxgl-samples` module together cover a good range of the basics, though some (new) features may lack examples and documentation.
